app/src/test/java/com/kickstarter/viewmodels/EditProfileViewModelTest.kt

```diff
@@ -0,0 +1,195 @@
+package com.kickstarter.viewmodels
+
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.MockCurrentUserV2
+import com.kickstarter.libs.utils.extensions.addToDisposable
+import com.kickstarter.mock.factories.UserFactory
+import com.kickstarter.mock.services.MockApiClientV2
+import com.kickstarter.models.User
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subscribers.TestSubscriber
+import org.junit.After
+import org.junit.Test
+
+class EditProfileViewModelTest : KSRobolectricTestCase() {
+
+    private lateinit var vm: EditProfileViewModel.EditProfileViewModel
+
+    private val hidePrivateProfileRow = TestSubscriber<Boolean>()
+    private val user = TestSubscriber<User>()
+    private val userAvatarUrl = TestSubscriber<String>()
+    private val userName = TestSubscriber<String>()
+    private val unableToSavePreferenceError = TestSubscriber<String>()
+
+    private val disposables = CompositeDisposable()
+
+    private fun setUpEnvironment(environment: Environment) {
+        this.vm = EditProfileViewModel.EditProfileViewModel(environment)
```

Comment:
Fyi, non mandatory, you can use the factory to create the VM, so it is also covered by the tests: 
`Factory(env).create(EditProfileViewModel::class.java)` you can take a look for example at `ProjectSocialViewModelTest:28`